src/core/mesh/qgstriangularmesh.h

```diff
@@ -118,6 +120,47 @@ class CORE_EXPORT QgsTriangularMesh
 
     QVector<QVector3D> vertexNormals( float vertScale ) const;
 
+    /**
+     * Returns simplified meshes. The first simplified mesh is simplified with a goal of a number of triangles equals to the
+     * number of triangles of the base mesh divised by the reduction factor. For the following mesh the same reduction factor is used with
+     * the prededent goal of number of triangles.
+     * There are as many simplified meshes as necessary to have a the minimum triangles count on the last simplified mesh.
+     *
+     * The caller has to take the ownership of returned meshes.
+     *
+     * \param reductionFactor is the factor used to reduce the number of triangles of the mesh
+     * \param minimumTrianglesCount is the minimal faces count on simplified mesh
+     * \since QGIS 3.14
+     */
+    QVector<QgsTriangularMesh *> simplifyMesh( double reductionFactor, int minimumTrianglesCount = 10 ) const;
```

Comment:
this function should probably be outside the triangular mesh base class in some utils class

Reply:
This method used directly private members of the based mesh and of the simplified mesh. Implement this function outside the class leads to implements setters/getters.

Reply:
is it private method?

Reply:
no, as it is used by the mesh layer.

Reply:
still, it does not fit to qgsTriangularMesh here in my opinion... can we make it private function in the QgsMeshLayer or somewhere in mesh layer utils?

Reply:
but if you insist on this location, I am open to keep it here